java/src/main/java/ai/rapids/cudf/Rmm.java

```diff
@@ -70,6 +70,9 @@ public static void initialize(int allocationMode, boolean enableLogging, long po
     }
   }
 
+  /** Return the amount of RMM memory allocated in bytes. */
+  public static native long getTotalMemoryAllocated();
```

Comment:
This should probably be documented somewhere (either here or the tracking adaptor) that the returned value is, at best, an estimate. There is no guarantee that when you ask for `n` bytes that the resource only allocates `n` bytes. You've accounted for one source of inaccuracy by accounting for `alignment`, but again, there's zero guarantee that this makes the tracking 100% accurate. The documented behavior of `device_memory_resource::allocate` is that it will return _at least_ `n` bytes.

Reply:
Excellent point.  Updated the Javadoc here to explain it's a lower-bound on the amount of memory allocated.